rules/qc.smk

```diff
@@ -0,0 +1,225 @@
+## RSEQC
+rule rseqc_gtf2bed12:
+    input:
+        config["ref"]["annotation"]
+    output:
+        "star/rseqc/annotation.bed"
+    log:
+        "logs/gtf2bed12.stderr"
+    params:
+        script = "../scripts/gtf2bed.pl"
+    shell:
+        """
+        perl {params.script} {input} > {output} 2> {log}
+        """
+
+
+rule rseqc_junction_annotation:
+    input:
+        "star/{sample}-{unit}/Aligned.out.bam",
+        "star/rseqc/annotation.bed"
+    output:
+        touch('star/rseqc/{sample}-{unit}.junction_annotation.done')
+    priority: 1
+    log:
+        "logs/rseqc/rseqc_junction_annotation/{sample}-{unit}.stdout",
+        "logs/rseqc/rseqc_junction_annotation/{sample}-{unit}.stderr"
+    params:
+        extra = r'-q 255',  # STAR uses 255 as a scrore for uniq mappers
+        prefix= 'star/rseqc/{sample}-{unit}.junctionanno'
+    conda:
+        "../envs/rseqc.yaml"
+    shell:
+        """
+        junction_annotation.py {params.extra} -i {input[0]} -r {input[1]} -o {params.prefix} > {log[0]} 2> {log[1]};
+        """
+
+        
+rule rseqc_junction_saturation:
+    input:
+        "star/{sample}-{unit}/Aligned.out.bam",
+        "star/rseqc/annotation.bed"
+    output:
+        touch('star/rseqc/{sample}-{unit}.junction_saturation.done')
+    priority: 1
+    log:
+        "logs/rseqc/rseqc_junction_saturation/{sample}-{unit}.stdout",
+        "logs/rseqc/rseqc_junction_saturation/{sample}-{unit}.stderr"
+    params:
+        extra = r'-q 255', 
+        prefix = 'star/rseqc/{sample}-{unit}.junctionsat'
+    conda:
+        "../envs/rseqc.yaml"
+    shell:
+        """
+        junction_saturation.py {params.extra} -i {input[0]} -r {input[1]} -o {params.prefix} > {log[0]} 2> {log[1]};
+        """
+
+
+rule rseqc_stat:
+    input:
+        "star/{sample}-{unit}/Aligned.out.bam",
+    output:
+        "star/rseqc/{sample}-{unit}.stats.txt"
+    priority: 1
+    log:
+        "logs/rseqc/rseqc_stat/{sample}-{unit}.stderr"
+    params:
+        extra = r''
+    conda:
+        "../envs/rseqc.yaml"
+    shell:
+        """
+        bam_stat.py {params.extra} -i {input} > {output} 2> {log}
+        """
+
+        
+rule rseqc_infer:
+    input:
+        "star/{sample}-{unit}/Aligned.out.bam",
+        "star/rseqc/annotation.bed"
+    output:
+        "star/rseqc/{sample}-{unit}.infer_experiment.txt"
+    priority: 1
+    log:
+        "logs/rseqc/rseqc_infer/{sample}-{unit}.stderr"
+    params:
+        extra = r''
```

Comment:
Please omit the extra params if they are not needed.

Reply:
Sure